transformer_engine/pytorch/softmax.py

```diff
@@ -260,7 +260,9 @@ def forward_fused_softmax(
     ) -> torch.Tensor:
         """Fused masked softmax kernel"""
         b, np, sq, sk = inp.size()
-        scale = self.scale if self.scale is not None else 1.0
+        scale = 1.0
+        if self.scale is not None and self.input_in_fp16:
```

Comment:
I don't like this - I would not want function named "scaled_softmax" to apply the scale I provided only in certain cases, this is surprising and will lead to bugs. It should be the responsibility of the caller of that softmax to either provide or not provide the scale depending on the input type.